connman/plugins/blacklist_monitor.c

```diff
@@ -0,0 +1,610 @@
+/*
+ *  Connection Manager
+ *
+ *  Copyright (C) 2022 Jolla Ltd. All rights reserved.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <errno.h>
+
+#define CONNMAN_API_SUBJECT_TO_CHANGE
+#include <connman/device.h>
+#include <connman/ipconfig.h>
+#include <connman/inet.h>
+#include <connman/log.h>
+#include <connman/network.h>
+#include <connman/plugin.h>
+#include <connman/service.h>
+
+#include "src/connman.h"
+
+#define IPV6_ANY "::"
+
+struct rtnl_gw {
+	int index;
+	char *gw;
+	int metric;
+};
+
+static GSList *rtnl_gw_list = NULL;
+static GSList *rtnl_gw_restore_list = NULL;
+
+static struct rtnl_gw *new_rtnl_gw(int index, const char *gw, int metric)
+{
+	struct rtnl_gw *item;
+
+	item = g_try_new0(struct rtnl_gw, 1);
+	if (!item) {
```

Comment:
This `g_try_new0` and the subsequent check of the return value doesn't make sense because e.g. `g_strdup()` below does `g_new` and doesn't check the return value. OOM condition is unrecoverable (because the checks are random and because of over-commit).

Reply:
I forgot that small allocations are let to fail as that is a bigger issue anyways when those fail. Changed in 6a6cd3e0d54063f0ee355c5c8ab76eff91b61895 just to use `g_new0` and ignore error, if if crashes it is not the only thing that is bound to crash in such case.